pkg/setting/setting_unified_alerting.go

```diff
@@ -63,26 +64,53 @@ type UnifiedAlertingSettings struct {
 	EvaluationTimeout              time.Duration
 	ExecuteAlerts                  bool
 	DefaultConfiguration           string
-	Enabled                        bool
+	Enabled                        *bool // determines whether unified alerting is enabled. If it is nil then user did not define it and therefore its value will be determined during migration. Services should not use it directly.
 	DisabledOrgs                   map[int64]struct{}
 }
 
+// IsEnabled returns true if UnifiedAlertingSettings.Enabled is either nil or true.
+// It hides the implementation details of the Enabled and simplifies its usage.
+// It is supposed to be used in background services.
+func (u *UnifiedAlertingSettings) IsEnabled() bool {
+	return u.Enabled == nil || *u.Enabled
+}
+
 // ReadUnifiedAlertingSettings reads both the `unified_alerting` and `alerting` sections of the configuration while preferring configuration the `alerting` section.
 // It first reads the `unified_alerting` section, then looks for non-defaults on the `alerting` section and prefers those.
 func (cfg *Cfg) ReadUnifiedAlertingSettings(iniFile *ini.File) error {
+	var err error
 	uaCfg := UnifiedAlertingSettings{}
 	ua := iniFile.Section("unified_alerting")
-	uaCfg.Enabled = ua.Key("enabled").MustBool(false)
-
-	// TODO: Deprecate this in v8.4, if the old feature toggle ngalert is set, enable Grafana 8 Unified Alerting anyway.
-	if !uaCfg.Enabled && cfg.FeatureToggles["ngalert"] {
+	enabled, err := ua.Key("enabled").Bool()
+	if err == nil {
+		uaCfg.Enabled = &enabled
+		// If unified alerting is disabled but legacy alerting is undefined, fallback to legacy enabled.
+		if AlertingEnabled == nil {
+			alertingEnabled := !enabled
+			AlertingEnabled = &alertingEnabled
+		}
+	} else if cfg.FeatureToggles["ngalert"] {
+		// If old feature toggle ngalert is set, enable unified alerting.
+		// TODO: Remove this in v9
 		cfg.Logger.Warn("ngalert feature flag is deprecated: use unified alerting enabled setting instead")
-		uaCfg.Enabled = true
-		AlertingEnabled = false
+		uaCfg.Enabled = new(bool)
+		*uaCfg.Enabled = true
+		AlertingEnabled = new(bool)
+	} else if AlertingEnabled != nil && !*AlertingEnabled {
+		// If unified alerting is not defined but legacy alerting is disabled, enable unified alerting.
+		enabled = true
+		uaCfg.Enabled = &enabled
+	} else {
+		uaCfg.Enabled = nil
```

Comment:
isn't the default value at this point `nil`?  I think this line might not be necessary.

Reply:
In my opinion, this helps readability because we rarely use a pointer to a value type. In this case else clause emphasizes this nuance.

Reply:
Sounds good, let's move it to the top then and start with the assumption that this `nil`.